### ISSUE TITLE
Per Batch Communication to Accomplish Synchronization 

### ISSUE THREAD

**Issue by psklavos1** [adap/flower#2593]
### What is your question?

The requirement is to send a custom metric at the end of each batch,(end of model.fit batch of each client). This metric is going to be used to calculate an aggregated sum that is going to be compared to a threshold. If the aggregated sum is greater than the threshold, then the round stops, we execute some updates, and a new round starts. This way sync, in a naive way, is achieved per batch of data sent. The part that troubles me most is how to communicate the metrics at each epoch blocking the fit execution of all the clients in order to be in sync.

Changing the aggregate_fit was my first move but I cannot find a way to communicate the metric per batch. I have concluded that I need a custom way to communicate with a metric server that is going to execute the synchronization overview, However i was wandering whether there is something already in the framework that can help me achieve this functionality

**Comment by jafermarq** [comment]
Hi @psklavos1 , so the idea is for the server/strategy to send a signal to the clients to (potentially) _break_ from the training loop in client's `fit()` if your condition is met ? For this the clients (while doing `fit()`) stream a metric to the server for it later decide to stop the training on the client side.

**Comment by jafermarq** [comment]
with a custom strategy that effectively treats N rounds as N local training steps, and only communicate the global model when N=0, you could implement this feature. This assume that for the N rounds the clients sampled are the same. That should work. Happy to elaborate in greater detail

The alternative, i.e., make the strategy send a message to clients to stop doing `fit()` and return isn't immediately obvious to me how to do it. Primarily because how how the [Server class](https://github.com/adap/flower/blob/5cf5c827abd5ba70cab0c9dce306733780920da6/src/py/flwr/server/server.py#L55) works.

**Comment by WilliamLindskog** [comment]
Hi @psklavos1

Thanks for raising this. Just checking in here. Are you still experiencing this issue? If no, can we close this?

**Comment by WilliamLindskog** [comment]
Hi again, 

this seems to have been resolved with above comments/suggestions. Closing this issue for now.
